frontend/src/components/nodes/NodesMap.tsx

```diff
@@ -0,0 +1,581 @@
+          `</div>` +
+          `<div>` +
+            `<div style="color: #ffffff; opacity: 0.4; font-size: 10px; line-height: 10px; letter-spacing: 0.2px; font-weight: bold; font-family: BwSeidoRound; padding: 0 0 6px">` + 'Last seen' + `</div>` +
+            `<div style="color: #ffffff; font-size: 14px; line-height: 14px; letter-spacing: 0.4px; font-weight: bold; font-family: BwSeidoRound;">` + moment.duration(moment().diff(moment(data.lastSeen))).as('seconds').toFixed() + 's ago' + `</div>` +
+          `</div>` +
+          `<div>` +
+            `<div style="color: #ffffff; opacity: 0.4; font-size: 10px; line-height: 10px; letter-spacing: 0.2px; font-weight: bold; font-family: BwSeidoRound; padding: 0 0 6px">` + 'Location' + `</div>` +
+            `<div style="color: #ffffff; font-size: 14px; line-height: 14px; letter-spacing: 0.4px; font-weight: bold; font-family: BwSeidoRound;">` + data.location + `</div>` +
+          `</div>` +
+        `</div>` +
+      `</div>`;
+  }
+
+  render() {
+    const map = <Datamap
+      responsive
+      geographyConfig={{
+        popupOnHover: false,
+        highlightOnHover: false,
+        borderColor: '#121314',
+      }}
+      fills={{
+        defaultFill: '#121314',
+        validatorBubbleFill: '#8DD4BD',
+        nonValidatorBubbleFill: '#8DD4BD',
+      }}
+      bubbles={this.state.nodesData}
+      pins={this.state.newNodes}
+      removedNodes={this.state.removedNodes}
+      nodeClusters={this.state.nodeClusters}
+      bubbleOptions={{
+        borderWidth: 1,
+        borderColor: '#1C1D1F',
+        fillOpacity: 0.6,
+        highlightFillOpacity: 1,
+        highlightFillColor: '#8DD4BD',
+        highlightBorderColor: '#8DD4BD',
+        highlightBorderWidth: 2,
+        animate: false,
+        popupTemplate: this.renderBubbleTooltip,
+      }}
+      pinOptions={{
+        borderWidth: 1,
+        borderColor: '#8DD4BD',
+        fillColor: '#8DD4BD',
+        fillOpacity: 0.6,
+        highlightFillOpacity: 1,
+        highlightFillColor: '#8DD4BD',
+        highlightBorderColor: '#8DD4BD',
+        highlightBorderWidth: 2,
+        radius: 4,
+      }}
+      removedNodesOptions={{
+        borderWidth: 1,
+        borderColor: '#FF585D',
+        fillColor: '#FF585D',
+        fillOpacity: 1,
+        highlightFillOpacity: 1,
+        highlightFillColor: '#FF585D',
+        highlightBorderColor: '#FF585D',
+        highlightBorderWidth: 2,
+        radius: 4,
+      }}
+      nodeClustersOptions={{
+        borderWidth: 1,
+        borderColor: '#1C1D1F',
+        fillColor: '#8DD4BD',
+        fillOpacity: 1,
+        highlightFillOpacity: 1,
+        highlightFillColor: '#8DD4BD',
+        highlightBorderColor: '#8DD4BD',
+        highlightBorderWidth: 2,
+        radius: 6,
+        popupTemplate: () => {
+          return `<div style="color: white">template</div>`
```

Comment:
It seems that this implementation is not finished.

![image](https://user-images.githubusercontent.com/304265/86114779-6417be00-baa1-11ea-95a8-99cdef968570.png)

Reply:
yeah, i pushed a WIP commit as a checkpoint, in case my laptop suddenly crashes to have the code saved :)